mail_show_follower/models/mail_mail.py

```diff
@@ -59,97 +59,31 @@ def remove_p(markup_txt):
-                        if cc_internal:
-                            partners = partners_obj.filtered(
-                                lambda x: x.id not in user_partner_ids
-                                and (
-                                    not x.user_ids
-                                    or any(x.mapped("user_ids.show_in_cc"))
-                                )
-                            )
-                        else:
-                            partners = partners_obj.filtered(
-                                lambda x: x.id not in user_partner_ids
-                                and (
-                                    not x.user_ids
-                                    or group_portal in x.user_ids.mapped("groups_id")
-                                )
-                            )
-                        partners = partners.filtered(
-                            lambda x: not x.user_ids
-                            or x.user_ids  # otherwise, email is not sent
-                            and "email" in x.user_ids.mapped("notification_type")
-                        )
-                        # set proper lang for recipients
-                        langs = list(
-                            filter(
-                                bool,
-                                mail.mapped("recipient_ids.lang")
-                                + [
-                                    mail.author_id.lang,
-                                    self.env.company.partner_id.lang,
-                                ],
-                            )
-                        )
-                        # get show follower text
-                        final_cc = mail.with_context(
-                            lang=langs and langs[0]
-                        )._build_cc_text(partners)
-                        # it is saved in the body_html field so that it does
-                        # not appear in the odoo log
-                        mail.body_html = final_cc + mail.body_html
+        group_user = self.env.ref("base.group_user")
+
+        for mail in self:
+            if not (mail.model and mail.res_id and group_user):
+                continue
+            # recipients from any Notification Type (i.e. email, inbox, etc.)
+            recipients = mail.notification_ids.res_partner_id
+            record = self.env[mail.model].browse(mail.res_id)
+            company = getattr(record, "company_id", self.env.company)
+            show_internal_users = company and company.show_internal_users_cc
+            show_in_cc_recipients = recipients._filter_shown_in_cc(show_internal_users)
+            if len(show_in_cc_recipients) <= 1:
+                continue
+
+            langs = (
+                mail.notification_ids.res_partner_id.mapped("lang")
+                or mail.author_id.lang
+                or company.partner_id.lang
+            )
+            final_cc = (
+                mail.with_context(lang=langs[0])
```

Comment:
issue: `langs` is of type `list[str] | str | bool` as per the code above. This is only supporting the `list[str]` case.

suggestion: I think the bug you guys discovered deserves at least a new test.